num.nix

```diff
@@ -153,4 +209,446 @@ rec {
 
     # TODO: polar, magnitude, phase
   };
+
+  /* truncate :: float -> int
+
+     Truncate a float to an int, rounding towards 0.
+  */
+  # may god have mercy on my soul.
+  truncate = f:
+    let chars = builtins.toJSON f;
+    in builtins.fromJSON (string.takeWhile (c: c != ".") chars);
+
+  /* floor :: float -> int
+
+     Floor a floating point number, rounding towards negative infinity.
+  */
+  floor = builtins.floor or (f: truncate (if f < 0 then f - 1 else f));
+
+  /* ceil :: float -> int
+
+     Ceiling a floating point number, rounding towards positive infinity.
+  */
+  ceil = builtins.ceil or (f: truncate (if f > 0 then f + 1 else f));
+
+  /* round :: float -> int
+
+     Round a floating-point number to the nearest integer, biased away from 0.
+  */
+  round = f: signum f * floor (abs f + 0.5);
+
+  /* tryParseInt :: string -> maybe int
```

Comment:
we now have 'Optional' and 'Nullable'. My default has been to return 'Optional' values over 'Nullable' ones because of the nicer nesting properties. I would prefer if this returned an 'Optional'. This goes for everything else in this file too. Note also that 'maybe' doesn't exist anymore. Also I think I forgot to make 'optional'/'nullable' lowercase in other docs, oops.

Reply:
this is a good point. I've changed `(try)?Parse(Int|Float)` accordingly.